app/controllers/api/v1/user_annotations_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Api
+  module V1
+    # API methods for user annotations
+
+    class UserAnnotationsController < ApiBaseController
+      include Concerns::Authenticator
+      include Concerns::StudyAware
+      include Concerns::ApiCaching
+      include Swagger::Blocks
+
+      before_action :set_current_api_user!
+      before_action :set_study
+      before_action :check_study_view_permission
+
+      swagger_path '/site/studies/{accession}/user_annotation' do
+        operation :post do
+          key :tags, [
+              'Site'
+          ]
+          key :summary, 'Create user annotation'
+          key :description, 'Create new custom user annotation for the study'
+          key :operationId, 'site_study_user_create_annotation_path'
+          parameter do
+            key :name, :accession
+            key :in, :path
+            key :description, 'Study accession number (e.g. SCPXXX)'
+            key :required, true
+            key :type, :string
+          end
+          parameter do
+            key :name, :user_annotation
+            key :in, :body
+            key :description, 'UserAnnotation object'
+            schema do
+              key :'$ref', :UserAnnotation
+            end
+          end
+          response 200 do
+            key :description, 'Successfully created user annotation'
+          end
+          extend SwaggerResponses::StudyControllerResponses
+        end
+      end
+
+      def create_user_annotation
+
+          message, annotations, status = UserAnnotationService.create_user_annotation(
+            @study, params[:name], params[:user_data_arrays_attributes],
+            params[:cluster], params[:loaded_annotation],
+            params[:subsample_threshold], current_api_user
+          )
+
+          response_body = {
+            message: message, annotations: annotations
+          }
+
+          render json: response_body, status: status and return
```

Comment:
The `and return` is not necessary here as calling `render` halts execution.  Usually, we only need `and return` when there is an if/else block that calls `redirect_to` as an escape condition to avoid executing any code further in the method.

Reply:
Done in 296cd75.